modules/kafka/src/test/java/com/devonfw/example/base/config/TestMessageSenderConfig.java

```diff
@@ -0,0 +1,249 @@
+package com.devonfw.example.base.config;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.Collections;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.kafka.core.ConsumerFactory;
+import org.springframework.kafka.core.DefaultKafkaConsumerFactory;
+import org.springframework.kafka.core.DefaultKafkaProducerFactory;
+import org.springframework.kafka.core.KafkaTemplate;
+import org.springframework.kafka.core.ProducerFactory;
+import org.springframework.kafka.test.EmbeddedKafkaBroker;
+
+import com.devonfw.module.kafka.common.messaging.api.config.KafkaCommonProperties;
+import com.devonfw.module.kafka.common.messaging.api.config.KafkaConsumerProperties;
+import com.devonfw.module.kafka.common.messaging.api.config.KafkaProducerProperties;
+import com.devonfw.module.kafka.common.messaging.api.config.MessageSenderProperties;
+import com.devonfw.module.kafka.common.messaging.health.api.config.KafkaHealthIndicatorProperties;
+import com.devonfw.module.kafka.common.messaging.health.impl.KafkaHealthIndicator;
+import com.devonfw.module.kafka.common.messaging.impl.MessageSenderImpl;
+import com.devonfw.module.kafka.common.messaging.logging.impl.MessageLoggingSupport;
+import com.devonfw.module.kafka.common.messaging.logging.impl.ProducerLoggingListener;
+import com.devonfw.module.kafka.common.messaging.trace.impl.MessageSpanInjector;
+import com.devonfw.module.kafka.common.messaging.util.KafkaPropertyMapper;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import brave.Tracer;
+import kafka.server.KafkaConfig;
+
+/**
+ * @author ravicm
+ *
+ */
+@Configuration
+public class TestMessageSenderConfig {
```

Comment:
Couldn't we provide more defaults in devon4j-kafka so that tests and real users to not need that many configurations?

Reply:
do you mean to provide configuration for Receiver test also ? this class has been instantiated only with required properties and rest all will be default. i will have another look reduce further more.

Reply:
No my point where not the tests. I was just wondering why so many beans have to be created. I assume that this has also to be done in a real application. Please check if is possible to provide a auto-configuration for default beans in devonfw-kafka

Reply:
Those default beans reads value from application.properties. we had major issue on reading port number of EmbeddedKafkaBroker. so we decided to keep seperate config class for tests.